### PR TITLE
Infer sticking from lyrics

### PR DESCRIPTION
**Pull request by miiizen**
This PR infers sticking from lyrics in musicxml files.  
There is one false positive in the test files.  I would have liked to confine this to percussion instruments, but import of percussion instruments is difficult and imperfect (as there are no standardised names).  Because of this, I've allowed inference as long as the instrument isn't a voice.

### CODE REVIEW

**Review by oktophonie — APPROVED** [review]

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 1]
```diff
@@ -3287,7 +3309,7 @@ void MusicXMLParserDirection::direction(const String& partId,
         tempoLine->setBeginText(simplifiedText);
         tempoLine->setContinueText(u"");
         m_inferredTempoLineStart = tempoLine;
-    } else if (isLikelySticking() && isPercussionStaff) {
+    } else if (isLikelySticking()) {
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3312]: Makes `bool isPercussionStaff = m_pass1.isPercussionStaff(partId);`further up useless/unneeded/creating a compiler warning

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 2]
```diff
@@ -7587,31 +7612,46 @@ void MusicXMLParserLyric::parse()
...
-        lyric->setPropertyFlags(Pid::COLOR, PropertyFlags::UNSTYLED);
+        item->setProperty(Pid::COLOR, lyricColor);
+        item->setPropertyFlags(Pid::COLOR, PropertyFlags::UNSTYLED);
     }
-    lyric->setVisible(printLyric);
+    item->setVisible(printLyric);
 
-    lyric->setPlacement(placement() == "above" ? PlacementV::ABOVE : PlacementV::BELOW);
-    lyric->setPropertyFlags(Pid::PLACEMENT, PropertyFlags::UNSTYLED);
+    item->setPlacement(placement() == "above" ? PlacementV::ABOVE : PlacementV::BELOW);
```
Comment (Jojo-Schmitz) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7630]: Unrelated to this PR, but isn't a
```cpp
    if (!placement.isEmpty()) {
        ...
    }
```
missing here?

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 1 above)
Comment (RomanPudashkin) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3312]: It's still used here:
handleDrumInstrument(isPercussionStaff, tick + m_offset);

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 3]
```diff
@@ -7539,6 +7563,7 @@ void MusicXMLParserLyric::parse()
     double relX = m_e.doubleAttribute("relative-x") * 0.1 * DPMM;
     m_relativeY = m_e.doubleAttribute("relative-y") * -0.1 * DPMM;
     m_defaultY = m_e.doubleAttribute("default-y") * -0.1 * DPMM;
+    LyricsSyllabic syllabic;
```
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7566]: It's better to assign something by default

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp [hunk 4]
```diff
@@ -7587,31 +7612,46 @@ void MusicXMLParserLyric::parse()
         return;
     }
 
+    TextBase* item;
```
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7615]: = nullptr

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.h [hunk 5]
```diff
@@ -141,9 +141,11 @@ struct GraceNoteLyrics {
 class MusicXMLParserLyric
 {
 public:
-    MusicXMLParserLyric(const LyricNumberHandler lyricNumberHandler, muse::XmlStreamReader& e, Score* score, MxmlLogger* logger);
+    MusicXMLParserLyric(const LyricNumberHandler lyricNumberHandler, muse::XmlStreamReader& e, Score* score, MxmlLogger* logger,
+                        bool isVoiceStaff);
     std::set<Lyrics*> extendedLyrics() const { return m_extendedLyrics; }
     std::map<int, Lyrics*> numberedLyrics() const { return m_numberedLyrics; }
+    std::vector<Sticking*> inferredStickings() const { return m_inferredStickings; }
```
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.h:148]: const std::vector<Sticking*>&

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 1 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:3312]: Ah, I missed that, thanks

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 4 above)
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7615]:
```suggestion
    TextBase* item = nullptr;
```

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp (hunk 3 above)
Comment (RomanPudashkin) [inline src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp:7566]:
```suggestion
    LyricsSyllabic syllabic = LyricsSyllabic::SINGLE;
```

File: src/importexport/musicxml/internal/musicxml/importmxmlpass2.h (hunk 5 above)
Comment (Jojo-Schmitz) [reply src/importexport/musicxml/internal/musicxml/importmxmlpass2.h:148]: You didn't apply this though?

**Pull request merged**